credsweeper/file_handler/text_content_provider.py

```diff
@@ -23,5 +24,13 @@ def get_analysis_target(self) -> List[AnalysisTarget]:
             list of analysis targets based on every row in file
 
         """
-        lines = Util.read_file(self.file_path)
+        if Util.get_extension(self.file_path) == ".xml":
+            try:
+                tree = ElementTree.parse(self.file_path)
+                lines = Util.get_xml_data(tree.getroot())
+            except ElementTree.ParseError:
+                lines = Util.read_file(self.file_path)
+        else:
+            lines = Util.read_file(self.file_path)
```

Comment:
```suggestion
        error = True
        if Util.get_extension(self.file_path) == ".xml":
            try:
                tree = ElementTree.parse(self.file_path)
                lines = Util.get_xml_data(tree.getroot())
                error = False
            except Exception as exc:
                logger.error(f"Parse xml fail:{exc}")
        if error:
            lines = Util.read_file(self.file_path)
```

Reply:
Thank you for suggestion!
Sadly `if error:` is not proper here... i think..
I'll refactor this code block.